src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -140,6 +141,29 @@ def get_dataset(
     return dataset
 
 
+@router.get("/me/datasets/{dataset_id}/metrics", response_model=Metrics)
+def get_current_user_dataset_metrics(
+    *,
+    db: Session = Depends(get_db),
+    dataset_id: UUID,
+    current_user: User = Security(auth.get_current_user),
+):
+    dataset = _get_dataset(db, dataset_id)
+
+    authorize(current_user, DatasetPolicyV1.get(dataset))
+
+    return {
+        "records": {
+            "count": datasets.count_records_by_dataset_id(db, dataset_id),
+        },
+        "responses": {
+            "count": datasets.count_responses_by_dataset_id_and_user_id(db, dataset_id, current_user.id),
```

Comment:
Do we really need this count? Normally, discarded + submitted = count. Once we introduce a new status, we can think about if we need a new `new_status` count or not.

Reply:
Yes we can avoid that count right now but I prefer to be explicit about it and don't involve status logic.